src/ui/public/vislib/lib/data.js

```diff
@@ -21,20 +21,20 @@ export default function DataFactory(Private) {
    * @param attr {Object|*} Visualization options
    */
   class Data {
-    constructor(data, attr, uiState) {
+    constructor(data, visConfigArgs, uiState) {
```

Comment:
Ideally `Data` would only consume the `visConfig` via it's public interface, rather than consuming it's args directly. Unfortunately there is a bit of a circular-dependency, with `Data` and `VisConfig` both needing each other in order to setup.

Do you think there is a way to prevent that? Maybe when `VisConfig` instantiates `Data` is passes down just the parameters it knows that `Data` needs?